packages/flutter/lib/src/material/ink_decoration.dart

```diff
@@ -0,0 +1,356 @@
+/// splashes to be drawn above the otherwise opaque graphics.
+///
+/// An alternative solution is to use a [MaterialType.transparency] material
+/// above the opaque graphics.
+///
+/// ## Limitations
+///
+/// This widget is subject to the same limitations as other ink effects, as
+/// described in the documentation for [Material]. Most notably, the position of
+/// an [Ink] widget must not change during the lifetime of the [Material] object
+/// unless a [LayoutChangedNotification] is dispatched each frame that the
+/// position changes. This is done automatically for [ListView] and other
+/// scrolling widgets, but is not done for animated transitions such as
+/// [SlideTransition].
+///
+/// ## Sample code
+///
+/// This example shows how a [Material] widget can have a yellow rectangle drawn
+/// on it using [Ink], while still having ink effects over the yellow rectangle:
+///
+/// ```dart
+/// new Material(
+///   color: Colors.teal[900],
+///   child: new Center(
+///     child: new Ink(
+///       color: Colors.yellow,
+///       width: 200.0,
+///       height: 100.0,
+///       child: new InkWell(
+///         onTap: () { /* ... */ },
+///         child: new Center(
+///           child: new Text('YELLOW'),
+///         )
+///       ),
+///     ),
+///   ),
+/// )
+/// ```
+///
+/// The following example shows how an image can be printed on a [Material]
+/// widget with an [InkWell] above it:
+///
+/// ```dart
+/// new Material(
+///   color: Colors.grey[800],
+///   child: new Center(
+///     child: new Ink.image(
+///       image: new AssetImage('cat.jpeg'),
+///       fit: BoxFit.cover,
+///       width: 300.0,
+///       height: 200.0,
+///       child: new InkWell(
+///         onTap: () { /* ... */ },
+///         child: new Align(
+///           alignment: Alignment.topLeft,
+///           child: new Padding(
+///             padding: const EdgeInsets.all(10.0),
+///             child: new Text('KITTEN', style: new TextStyle(fontWeight: FontWeight.w900, color: Colors.white)),
+///           ),
+///         )
+///       ),
+///     ),
+///   ),
+/// )
+/// ```
+///
+/// See also:
+///
+///  * [Container], a more generic form of this widget which paints itself,
+///    rather that defering to the nearest [Material] widget.
+///  * [InkDecoration], the [InkFeature] subclass used by this widget to paint
+///    on [Material] widgets.
+///  * [InkWell] and [InkResponse], which also draw on [Material] widgets.
+class Ink extends StatefulWidget {
+  /// Paints a decoration (which can be a simple color) on a [Material].
+  ///
+  /// The [height] and [width] values include the [padding].
+  ///
+  /// The `color` argument is a shorthand for `decoration: new
+  /// BoxDecoration(color: color)`, which means you cannot supply both a `color`
+  /// and a `decoration` argument. If you want to have both a `color` and a
+  /// `decoration`, you can pass the color as the `color` argument to the
+  /// `BoxDecoration`.
+  Ink({
```

Comment:
This is a nice short name. It's also a bit of a misnomer if we're going to say that splashes and ripples are drawn with "ink". A longer more horrible but more apt name would be MaterialAncestorDecorator.

Reply:
as we discussed, i agree that the name is a bit sketchy but i think we want this to be more attractive than Container or BoxDecoration or even Image.